eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -36,11 +36,12 @@ jobs:
       vmImage: '$(OSVmImage)'
 
     steps:
-
-      - template: ../../../common/pipelines/templates/steps/verify-links.yml
-        parameters:
-          Directory: ''
-          CheckLinkGuidance: $true
+      - ${{ if eq(variables['Build.Reason'], 'PullRequest') }}:
+        - template: /eng/common/pipelines/templates/steps/verify-links.yml
+          parameters:
+            Directory: ''
+            Urls: (git diff origin/${env:SYSTEM_PULLREQUEST_TARGETBRANCH} HEAD --name-only -- *.md)
```

Comment:
As we talked in the cpp repo we shouldn't add this logic unless we have an aggregate report type pipeline that we can use to check all the files. For this repo I suspect we should just keep it always checking all the MD files as it is a smaller repo.